File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        if(root==NULL) return result;
        queue<TreeNode*> q;
        q.push(root);
        int count=1;
        while(!(q.empty())){
            int i=count;
            count=0;
            vector<int> sub_result;
            while(i!=0){
            TreeNode *e=q.front();
            q.pop();
            i--;
            sub_result.push_back(e->val);
            if(e->left!=NULL){
                count++;
                q.push(e->left);
            }
            if(e->right!=NULL){
                count++;
                q.push(e->right);
            }
            }
            result.push_back(sub_result);
        }
        return result;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result=new ArrayList<List<Integer>>();
        if(root==null) return result;
        Queue<TreeNode> q=new LinkedList<>();
        q.add(root);
        int count=1;
        while(!(q.isEmpty())){
            List<Integer> sub_result=new ArrayList<Integer>();
            int e=count;
            count=0;
            while(e!=0){
                TreeNode t=q.remove();
                e--;
                sub_result.add(t.val);
                if(t.left!=null){
                    count++;
                    q.add(t.left);
                }
                if(t.right!=null){
                    count++;
                    q.add(t.right);
                }
            }
                result.add(sub_result);
        }
        return result;
    }
}